src/Chainweb/Pact/PactService.hs

```diff
@@ -815,3 +826,18 @@ transactionsFromPayload plData = do
     return $! V.fromList theRights
   where
     toCWTransaction bs = codecDecode chainwebPayloadCodec bs
+
+execLookupPactTxs
+    :: PayloadCas cas
+    => T2 BlockHeight BlockHash
+    -> Vector P.PactHash
+    -> PactServiceM cas (Vector (Maybe (T2 BlockHeight BlockHash)))
+execLookupPactTxs (T2 leafHeight leafHash) txs =
+    withDiscardedBatch $
+    withCheckpointerRewind (Just (leafHeight + 1, leafHash))
```

Comment:
What's the justification for `leafHeight + 1`?  Do you mind commenting your thinking with these calls? (re: discardedBatch, rewind to specific height, where you think the bheight + bhash pair are coming from etc)

Reply:
This is just the way the api works -- it's documented in the outer interface but not in here. The functions here:

`withDiscardedBatch` sets a sqlite "Batch" savepoint, runs the action, then discards to "Batch", effectively making the change a no-op in terms of db mutation (another important reason to batch the requests) -- but maybe we don't care about moving the restore point here because we're moving it to the cut value and that's a good place for it to be. I'll remove the call

Reply:
@gregorycollins it might be worth making a version of `withCheckpointerRewind` that does the `succ` on the blockHeight to document this behavior; as it is we're manually doing it in three places -- here, `execLocal`, and `execNewBlock` which is creaky.